File: 30daysOFcode_Php/day 12.php
<?php

class person {
    protected $firstName, $lastName, $id;
    
    public function __construct($first_name, $last_name, $identification) {
        $this->firstName = $first_name;
        $this->lastName = $last_name;
        $this->id = $identification;
    }
    
    function printPerson() {
        print("Name: {$this->lastName}, {$this->firstName}\n");
        print("ID: {$this->id}\n");
    }
}

class Student extends person {
    
    private $testScores;
    public $firstName;
    public $lastName;
    public $id;
    public $scores;

    public function __construct(
        
        string $firstName,
        string $lastName,
        int $id,
        
        array $scores)
        {
            parent::__construct($firstName, $lastName, $id);
            $this->firstName = $firstName;
            $this->lastName = $lastName;
            $this->id = $id;
            $this->scores = $scores;
        }
                
    /*    
    *   Function Name: calculate
    *   Return: A character denoting the grade.
    */
    public function calculate() : string
    {
        $avg = array_sum($this->scores) / count($this->scores);

        if ($avg >= 90 && $avg <= 100) {
            return 'O';
        } elseif ($avg >= 80 && $avg < 90) {
            return 'E';
        } elseif ($avg >= 70 && $avg < 80) {
            return 'A';
        } elseif ($avg >= 55 && $avg < 70) {
            return 'P';
        } elseif ($avg >= 40 && $avg < 55) {
            return 'D';
        } else {
            return 'T';
        }
    }
}


$file = fopen("php://stdin", "r");

$name_id = explode(' ', fgets($file));

$first_name = $name_id[0];
$last_name = $name_id[1];
$id = (int)$name_id[2];

fgets($file);

$scores = array_map(intval, explode(' ', fgets($file)));

$student = new Student($first_name, $last_name, $id, $scores);

$student->printPerson();

print("Grade: {$student->calculate()}");

File: 30daysOFcode_Swift/Day 12.swift
// Class Student
class Student: Person {
	var testScores: [Int]
	
	// Write the Student class initializer
	init(firstName: String, lastName: String, identification: Int, scores: [Int]) {
		self.testScores = scores;
		super.init(firstName: firstName, lastName: lastName, identification: identification);
	}
	
	// Write the calculate method
	func calculate() -> String {
		var average = 0;
		for i in testScores {
			average += i;
		}
		average = average / testScores.count;

		if(average >= 90) {
			return "O"; // Outstanding
		}
		else if(average >= 80) {
			return "E"; // Exceeds Expectations
		}
		else if(average >= 70) {
			return "A"; // Acceptable
		}
		else if(average >= 55) {
			return "P"; // Poor
		}
		else if(average >= 40) {
			return "D"; // Dreadful
		}
		else {
			return "T"; // Troll
		}
	}
	
} // End of class Student
